JsonSchema/SchemaKeywordRegistry.cs

```diff
@@ -18,40 +20,126 @@ namespace Json.Schema;
 public static class SchemaKeywordRegistry
 {
 	private static readonly ConcurrentDictionary<string, Type> _keywords;
+	private static readonly ConcurrentDictionary<Type, JsonTypeInfo> _keywordTypeInfos;
 	private static readonly ConcurrentDictionary<Type, IJsonSchemaKeyword> _nullKeywords;
+	internal static bool RequiresDynamicSerialization { get; private set; }
```

Comment:
I don't think this needs to be set globally.  Can we make it per keyword and have a method that gets whether the particular keyword needs to be dynamic?

Reply:
I can scope it to just the dynamic keywords.

Reply:
No, that wouldn't catch custom keywords.  If I get complaints, I'll change it.